### PR TITLE
Add an nginx based HTTP redirector

### PR DESCRIPTION
**Pull request by yuvipanda**
Generic enough that we can use this later for other redirects too.

### CODE REVIEW

File: config/prod.yaml [hunk 1]
```diff
@@ -13,3 +13,10 @@ binderhub:
       host: hub.mybinder.org
 
   googleAnalyticsCode: "UA-101904940-1"
+
+redirector:
+  redirects:
+    - type: host
+      host:
+        from: beta.mybinder.org
```
Comment (choldgraf) [inline config/prod.yaml]: shouldn't this just point to `mybinder.org`?

Comment (yuvipanda) [reply config/prod.yaml]: good catch

Comment (yuvipanda) [reply config/prod.yaml]: Fixed

File: mybinder/templates/redirector/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+metadata:
+  name: redirector
+  labels:
+    app: redirector
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    kubernetes.io/tls-acme: "true"
+spec:
+  rules:
+    {{ range $redirect := .Values.redirector.redirects }}
```
Comment (willingc) [inline mybinder/templates/redirector/ingress.yaml]: If I run `helm lint --debug --home './config/' mybinder/` on the PR, I get the following error for this line:
```
==> Linting mybinder/
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: render error in "mybinder/templates/redirector/ingress.yaml": template: mybinder/templates/redirector/ingress.yaml:12:33: executing "mybinder/templates/redirector/ingress.yaml" at <.Values.redirector.r...>: can't evaluate field redirects in type interface {}
```

Comment (yuvipanda) [reply mybinder/templates/redirector/ingress.yaml]: Did you pass in -f config/prod.yaml too? redirects are defined only in the prod / staging specific files...

I'll add a shim to values.yaml so this continues to work, however!

**Pull request merged**